src/writer.rs

```diff
@@ -40,35 +44,51 @@ impl<F: LurkField> Write<F> for ContPtr<F> {
         if let Some(cont) = store.fetch_cont(self) {
             cont.fmt(store, w)
         } else {
-            Ok(())
+            Err(io::Error::new(
+                io::ErrorKind::Other,
+                "Cannot find continuation pointer",
+            ))
         }
     }
 }
 
 fn write_symbol<F: LurkField, W: io::Write>(
     w: &mut W,
-    store: &Store<F>,
-    sym: &Sym,
+    _store: &Store<F>,
+    sym: &Symbol,
 ) -> io::Result<()> {
-    let package = &store.lurk_package;
-    let maybe_abbr = package.relative_abbreviation(sym);
-    let symbol_name = maybe_abbr.full_name();
-    write!(w, "{symbol_name}")
+    let lurk_syms = Symbol::lurk_syms();
+    if let Some(sym) = lurk_syms.get(sym) {
+        write!(w, "{}", sym)
+    } else {
+        write!(w, "{}", sym)
```

Comment:
@johnchandlerburnham I think this is where you need the assertion. That is, I think this is the point at which a top-level symbol sharing a name with a Lurk symbol will get printed incorrectly (even though such a symbol cannot currently be easily produced by reading).